nomos-services/time/src/backends/ntp/mod.rs

```diff
@@ -0,0 +1,122 @@
+mod async_client;
+
+use crate::backends::common::slot_timer;
+use crate::backends::ntp::async_client::{AsyncNTPClient, NTPClientSettings};
+use crate::backends::TimeBackend;
+use crate::{EpochSlotTickStream, SlotTick};
+use cryptarchia_engine::{EpochConfig, Slot, SlotConfig};
+use futures::{Stream, StreamExt};
+use sntpc::{fraction_to_nanoseconds, NtpResult};
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::Duration;
+use time::OffsetDateTime;
+use tokio::time::{interval, MissedTickBehavior};
+use tokio_stream::wrappers::IntervalStream;
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct NtpSettings {
+    /// Ntp server address
+    pub ntp_server: String,
+    /// Ntp server settings
+    pub ntpclient_settings: NTPClientSettings,
+    /// Interval for the backend to contact the ntp server and update its time
+    pub update_interval: Duration,
+    /// Slot settings in order to compute proper slot times
+    pub slot_config: SlotConfig,
+    /// Epoch settings in order to compute proper epoch times
+    pub epoch_config: EpochConfig,
+    /// Base period length related to epochs, used to compute epochs as well
+    pub base_period_length: u64,
+}
+pub struct Ntp {
+    settings: NtpSettings,
+    client: AsyncNTPClient,
+}
```

Comment:
This is not tested. But as this PR went out of hand with changes already....it will be worked in a different pr.